transmogrifier/sources/xml/springshare.py

```diff
@@ -99,8 +105,10 @@ def get_source_link(
         link.
 
         Args:
+            source_record_id: Source record ID.
             source_base_url: Source base URL.
-            source_record_id: Record identifier for the source record.
-            xml: A BeautifulSoup Tag representing a single XML record.
+            source_record: A BeautifulSoup Tag representing a single OAI DC record in XML.
         """
-        return str(xml.find("dc:identifier").string)
+        if source_link := source_record.find("dc:identifier", string=True):
+            return str(source_link.string)
+        return ""
```

Comment:
Previously, the method would throw an attribute error if `dc:identifier` did not appear in the XML: 

```
AttributeError: 'NoneType' object has no attribute 'string'
```

Given that `source_link` is a required field, raising an error seems to be the right approach, but is there a cleaner way to handle this case?

Reply:
I think your instinct to use `SkippedRecordEvent` here is correct, @ghukill your thoughts?

Reply:
Agreed.  If we can't pull a `dc:identifier` value, then we don't have a source link to send people to, and it feels like it's not worth keeping the record (punting the awareness of this to later observability/metrics discussions).  

So yes, raising a `SkippedRecordEvent` feels like the right move!